server/services/shipTransfer.ts

```diff
@@ -42,36 +42,26 @@ export default class ShipTransferService {
         star.shipsActual! += shipsToTransfer;
         star.ships = Math.floor(star.shipsActual!);
 
-        // Generate an array of all requires DB updates.
-        let dbWrites: any[] = carriersAtStar.map(c => {
-            return {
-                updateOne: {
-                    filter: {
-                        _id: game._id,
-                        'galaxy.carriers._id': c._id
-                    },
-                    update: {
-                        'galaxy.carriers.$.ships': c.ships
+        if (writeToDB) {
+            // Generate an array of all requires DB updates.
+            let dbWrites: any[] = carriersAtStar.map(c => {
+                return {
+                    updateOne: {
+                        filter: {
+                            _id: game._id,
+                            'galaxy.stars._id': star._id
+                        },
+                        update: {
+                            'galaxy.stars.$.shipsActual': star.shipsActual,
+                            'galaxy.stars.$.ships': star.ships
+                        }
                     }
-                }
-            };
-        });
-
-        dbWrites.push({
```

Comment:
Why was this removed? Looks like this is the bit that updates the star's ship counts.

Reply:
Looks correct to me. It should do the exact same thing like before, just wrapped into an if statement. Or did I miss something?